adcm_aio_client/core/objects/cm.py

```diff
@@ -110,3 +120,78 @@ def get_own_path(self: Self) -> Endpoint:
 
 class ComponentsNode(NonPaginatedChildAccessor[Service, Component, None]):
     class_type = Component
+
+
+class Prototype[ParentObject: InteractiveObject](InteractiveChildObject[ParentObject]):
+    @property
+    def id(self: Self) -> int:
+        return int(self._data["id"])
+
+    @property
+    def name(self: Self) -> str:
+        return str(self._data["name"])
+
+    @property
+    def display_name(self: Self) -> str:
+        return str(self._data["displayName"])
+
+    @property
+    def description(self: Self) -> str:
+        return str(self._data["description"])
+
+    @property
+    def type(self: Self) -> ObjectType:
+        return ObjectType(self._data["type"])
+
+    @property
+    def version(self: Self) -> str:
+        return str(self._data["version"])
+
+    @property
+    def license(self: Self) -> dict:
+        return self._data["license"]
+
+    @cached_property
+    async def bundle(self: Self) -> Bundle:
+        bundle_id = self._data["bundle"]["id"]
+        response = await self._requester.get("bundles", bundle_id)
+
+        return self._construct(what=Bundle, from_data=response.as_dict())
+
+    def get_own_path(self: Self) -> Endpoint:
+        return (*self._parent.get_own_path(), "prototypes", self.id)
+
+
+class PrototypeNode[ParentObject: InteractiveObject](PaginatedChildAccessor[ParentObject, Prototype, None]):
+    class_type = Prototype
+
+
+class HostProvider(Deletable, InteractiveObject):
+    # data-based properties
+
+    @property
+    def id(self: Self) -> int:
+        return int(self._data["id"])
+
+    @property
+    def name(self: Self) -> str:
+        return str(self._data["name"])
+
+    @property
+    def description(self: Self) -> str:
+        return str(self._data["description"])
+
+    @property
+    def is_upgradable(self: Self) -> str:
+        return str(self._data["isUpgradable"])
+
+    @property
+    def main_info(self: Self) -> str:
+        return str(self._data["mainInfo"])
+
+    def get_own_path(self: Self) -> Endpoint:
+        return "hostproviders", self.id
+
+    @cached_property
+    def prototype(self: Self) -> "PrototypeNode":
+        return PrototypeNode(parent=self, path=(*self.get_own_path(), "prototypes"), requester=self._requester)
```

Comment:
Please check out requirements for the fields that should be available:
- id: int
- name: str
- display_name: str
- description: str

Reply:
@a-alferov What about `HostProvider` / `Provider` naming? Requirements specify `HostProvider`, so we follow it?

@Starovoitov Please also add `HostProvidersNode` (note plural form in nodes naming)

Reply:
deleted

Reply:
`main_info`, `is_upgradable` shouldn't be publicly accepted

Reply:
deleted